### PR TITLE
fix: Added Parameter ConditionalAccessPolicy to Resource SPOAccessControlSettings

### PR DESCRIPTION
**Pull request by hosebei**
#### Pull Request (PR) description
Added Parameter ConditionalAccessPolicy to Resource SPOAccessControlSettings. Based on the PNP Module documentation:
https://pnp.github.io/powershell/cmdlets/Set-PnPTenant.html#-conditionalaccesspolicy

#### This Pull Request (PR) fixes the following issues
None

### CODE REVIEW

**Comment by hosebei** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof [hunk 1]
```diff
@@ -22,4 +22,5 @@ class MSFT_SPOAccessControlSettings : OMI_BaseResource
     [Write, Description("Path to certificate used in service principal usually a PFX file.")] String CertificatePath;
     [Write, Description("Thumbprint of the Azure Active Directory application's authentication certificate to use for authentication.")] String CertificateThumbprint;
     [Write, Description("Managed ID being used for authentication.")] Boolean ManagedIdentity;
+    [Write, Description("Blocks or limits access to SharePoint and OneDrive content from un-managed devices.")] Boolean ConditionalAccessPolicy;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof:25]: The methods signature define it as String with a list of accepted values, whereas the schema defines it as Boolean. Can we please make sure the schema is correct and that it lists the  proper ValueMap and Values list. Thanks

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof (hunk 1 above)
Comment (hosebei) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.schema.mof]: thank you for pointing that out. I did miss that part to correctly include. I hope my fix does correct his addressed issue?

**Comment by hosebei** [comment]
Hello, I'm unsure If I did something wrong with my second commit to correct the issues outlined by @NikCharlebois .
Or if I just need to wait, until this get checked again. Thanks a lot for a response in advance.

**Comment by andikrueger** [comment]
Could you please add a changelog entry within the unrelease section and outline this addition? Thanks!

**Comment by hosebei** [comment]
Hello, I have added the parameter to the unreleased section. Now I'm not sure if I did something wrong, since it shows a conflict.
I apologize for any inconvenience.

**Comment by hosebei** [comment]
I have added the parameter to the unreleased section. I hope I did no mistake. Since I do not know how to resolve the Conflict furthermore.

**Comment by andikrueger** [comment]
I just resolved the conflict. Thanks for looking into it and your contribution!

**Comment by hosebei** [comment]
Thank you very much for correcting the conflict. I was unable to do it correctly.

**Comment by hosebei** [comment]
Hello @NikCharlebois , is there more to deliver, or can this be merged now?
Please tell me if I can adjust something to smoothen the process.

**Comment by hosebei** [comment]
hello @andikrueger , sorry to bother you again. since there was created a new release, the addition from this MR to the unreleased section does have a conflict again. I kindly ask you to resolve this conflict, since I think my changes would worsen the current situation.
do you have an idea why this MR was not included in the last version?

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1 [hunk 2]
```diff
@@ -413,7 +429,8 @@ function Test-TargetResource
             'DisallowInfectedFileDownload', `
             'ExternalServicesEnabled', `
             'EmailAttestationRequired', `
-            'EmailAttestationReAuthDays')
+            'EmailAttestationReAuthDays',
+        'ConditionalAccessPolicy')
```
Comment (andikrueger) [inline Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1:433]: Just one minor thing: Could you please make sure the indentation is the same as the ones above?

**Review by andikrueger — CHANGES_REQUESTED** [review]
Just one minor comment.

File: Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1 (hunk 2 above)
Comment (hosebei) [reply Modules/Microsoft365DSC/DSCResources/MSFT_SPOAccessControlSettings/MSFT_SPOAccessControlSettings.psm1]: I'm so sorry that my changes had so many issues. I have corrected the mentioned indentation.

**Comment by hosebei** [comment]
Hi @andikrueger please review my recent changes. :)

**Comment by hosebei** [comment]
hello @andikrueger and @NikCharlebois, is there something that hinders this addition to the dev tree?

**Review by andikrueger — APPROVED** [review]

**Comment by hosebei** [comment]
Hi @NikCharlebois since your approval is required too, can you approve the merge request as well? thank you very much

**Comment by hosebei** [comment]
hi @andikrueger, i allowing myself to kindly ask you, if there is something to provide, so nik could approve this change?

**Pull request merged**